src/EventStore.Core/Services/Transport/Grpc/Enumerators.AllSubscription.cs

```diff
@@ -384,28 +390,15 @@ await foreach (var @event in liveEvents.Reader.ReadAllAsync(_cancellationToken)
 					Unsubscribe();
 
 					liveEvents.Writer.Complete();
-
-					Fail(RpcExceptions.Timeout(msg));
-				}
-
-				void Fail(Exception exception) {
-					this.Fail(exception);
-					caughtUpSource.TrySetException(exception);
+					
+					await _channel.Writer.WriteAsync(new TimeOutError(msg), _cancellationToken).ConfigureAwait(false);
+					CloseChannel();
 				}
 			}
 
 			private ValueTask ConfirmSubscription() => Interlocked.CompareExchange(ref _subscriptionStarted, 1, 0) != 0
 				? new ValueTask(Task.CompletedTask)
-				: _channel.Writer.WriteAsync(new ReadResp {
-					Confirmation = new ReadResp.Types.SubscriptionConfirmation {
-						SubscriptionId = SubscriptionId
-					}
-				}, _cancellationToken);
-
-			private void Fail(Exception exception) {
-				Interlocked.Exchange(ref _subscriptionStarted, 1);
-				_channel.Writer.TryComplete(exception);
```

Comment:
this closes the channel with an exception but most of that places that used to call this now close the channel cleanly or dont close it at all, is that an intentional change?